eth/tracers/tracer.go

```diff
@@ -649,41 +669,68 @@ func (jst *Tracer) CaptureEnd(output []byte, gasUsed uint64, t time.Duration, er
 	}
 }
 
-// GetResult calls the Javascript 'result' function and returns its value, or any accumulated error
-func (jst *Tracer) GetResult() (json.RawMessage, error) {
-	// Transform the context into a JavaScript object and inject into the state
-	obj := jst.vm.PushObject()
-
-	for key, val := range jst.ctx {
-		switch val := val.(type) {
-		case uint64:
-			jst.vm.PushUint(uint(val))
-
-		case string:
-			jst.vm.PushString(val)
+func (jst *Tracer) CaptureEnter(typ vm.OpCode, from common.Address, to common.Address, input []byte, gas uint64, value *big.Int) {
+	if !jst.traceCallFrames {
+		return
+	}
+	if jst.err != nil {
+		return
+	}
+	// If tracing was interrupted, set the error and stop
+	if atomic.LoadUint32(&jst.interrupt) > 0 {
+		jst.err = jst.reason
+		return
+	}
 
-		case []byte:
-			ptr := jst.vm.PushFixedBuffer(len(val))
-			copy(makeSlice(ptr, uint(len(val))), val)
+	// Transform the frame into a JavaScript object and inject into the state
+	obj := jst.vm.PushObject()
+	jst.addToObj(obj, "type", typ.String())
+	jst.addToObj(obj, "from", from)
+	jst.addToObj(obj, "to", to)
+	jst.addToObj(obj, "input", input)
+	jst.addToObj(obj, "gas", gas)
+	if value != nil {
+		jst.addToObj(obj, "value", value)
+	}
+	jst.vm.PutPropString(jst.stateObject, "frame")
```

Comment:
Don't know how much speed is an issue, but the regular tracer reuses the same object, and re-points the accessors. So instead of setting `addToObj` on the duktape object, it just swaps a pointer. 

The downside is that insead of doing `foo.gas`, you need to do `foo.getGas()`. Then the work is deferred until it's actually used. 

If we ever at some point want to do that, we should do it from the start, since it requires modifying the js code that accesses the fields..

Reply:
Updated to use a similar approach as `step`. All `frame` and `frameResult` fields have to be now accessed via methods.